packages/theme/src/cli/services/metafields-pull.ts

```diff
@@ -161,5 +161,30 @@ function writeMetafieldDefinitionsToFile(path: string, content: unknown) {
   mkdirSync(shopifyDirectory)
 
   const filePath = joinPath(shopifyDirectory, 'metafields.json')
-  writeFileSync(filePath, JSON.stringify(content, null, 2))
+  const fileContent = JSON.stringify(content, null, 2)
+
+  if (!fileExistsSync(filePath)) {
+    addToGitIgnore(path, '.shopify')
+  }
+
+  writeFileSync(filePath, fileContent)
+}
+
+function addToGitIgnore(root: string, entry: string) {
```

Comment:
We should consider centralizing this logic in cli-kit 😅 
https://github.com/Shopify/hydrogen/blob/6bc6068fbffebbcd07c3d1db97956248bfa6a742/packages/cli/src/lib/shopify-config.ts#L135

Reply:
yes! can you open a PR for that please?

Reply:
I will do that in a following PR! Great call, @frandiox @isaacroldan! :)